Userland/Libraries/LibGfx/ImageFormats/JPEGXLLoader.cpp

```diff
@@ -1423,6 +1465,44 @@ struct ModularData {
                     nb_meta_channels += 2 - tr.begin_c;
                 else
                     nb_meta_channels += 1;
+            } else if (tr.tr == TransformInfo::TransformId::kSqueeze) {
+                if (tr.sp.is_empty())
+                    TRY(add_default_squeeze_params(tr, channel_infos, nb_meta_channels));
+
+                // "Let begin = sp[i].begin_c and end = begin + sp[i].num_c − 1.
+                // The channel list is modified as specified by the following code:"
+                for (u32 i = 0; i < tr.sp.size(); i++) {
+                    auto begin = tr.sp[i].begin_c;
+                    auto end = begin + tr.sp[i].num_c - 1;
+                    auto r = tr.sp[i].in_place ? end + 1 : channel_infos.size();
+                    if (begin < nb_meta_channels) {
+                        /* sp[i].in_place is true */
+                        /* end < nb_meta_channels */
+                        nb_meta_channels += tr.sp[i].num_c;
+                    }
+                    for (u32 c = begin; c <= end; c++) {
+                        auto w = channel_infos[c].width;
+                        auto h = channel_infos[c].height;
+                        /* w > 0 and h > 0 */
+
+                        ChannelInfo residu;
+                        if (tr.sp[i].horizontal) {
+                            channel_infos[c].width = (w + 1) / 2;
```

Comment:
This is also ceil_div, but I suppose if it's in the spec like that…

Reply:
Yep, also comes from the spec